src/node_metadata.h

```diff
@@ -38,7 +38,9 @@ namespace node {
   V(nghttp2)                                                                   \
   V(napi)                                                                      \
   V(llhttp)                                                                    \
-  V(uvwasi)
+  V(uvwasi)                                                                    \
+  V(acorn)                                                                     \
+  V(undici)
```

Comment:
Should we remove this from the list if node is built with `./configure --shared-builtin-undici/undici-path NODE_SHARED_BUILTIN_UNDICI_UNDICI_PATH`? See https://github.com/nodejs/node/pull/44376 to know more about how this can be externalized.

Reply:
Hi, so I went through that PR and was able to get some rudimentary understanding (hence apologize if anything I say is wrong, pls do correct me 😅) basically we would be able to build nodejs by providing our own dependency paths so I guess what we could do it add update the files storing the metadata versions when the `BuiltinLoader::AddExternalizedBuiltin(const char* id, const char* filename)` is called? in `src/node_builtins.cc` is that something feasible?

Reply:
> update the files storing the metadata versions when the `BuiltinLoader::AddExternalizedBuiltin(const char* id, const char* filename)` is called?

Unfortunately, that's not gonna work because it's gonna change the value on disk in the header file but the version is gonna stay the same in the running node process. I think a better approach would be to update the header file when the build is configured with that option but I think for now removing the versions of externalized dependencies too would be okay.

Reply:
Another option would be to parse the version file and update `Metadata::Versions::dependency_name` inside `AddExternalizedBuiltin()` but that's probably gonna be slow.

Reply:
>  I think a better approach would be to update the header file when the build is configured with that option 
Could you guide where I could make the code changes for this?

> but I think for now removing the versions of externalized dependencies too would be okay. 
As in removing it from node_metadata.h?

Reply:
I guess we could have something like excluding undici version if NODE_SHARED_BUILTIN_UNDICI_UNDICI_PATH is there then undici is excluded

Reply:
> Could you guide where I could make the code changes for this?

I think that would be in https://github.com/nodejs/node/blob/167d7a9ee88dd52fe882a5c1c87d8ddad897b0c3/configure.py#L2020-L2025

> As in removing it from node_metadata.h?

Yea, like
```cc
#ifndef NODE_SHARED_BUILTIN_UNDICI_UNDICI_PATH
  // declare and use the undici version variable
#endif  // NODE_SHARED_BUILTIN_UNDICI_UNDICI_PATH
```

> could we safely assume that a package.json would be there?

I don't think that's safe to assume because users can remove that file before running configure. In that case, I think not having the version info for that dependency would be acceptable.

> I guess we could have something like excluding undici version if NODE_SHARED_BUILTIN_UNDICI_UNDICI_PATH is there then undici is excluded

Exactly!

Reply:
Understood! thanks for the help could you say if NODE_SHARED_BUILTIN_UNDICI_UNDICI_PATH would be accessible from the node_metadata.h file? i couldn't find it defined in a header file

Reply:
Have updated @RaisinTen